planner/core/expression_rewriter.go

```diff
@@ -303,12 +303,25 @@ func (er *expressionRewriter) Enter(inNode ast.Node) (ast.Node, bool) {
 		}
 		er.ctxStack = append(er.ctxStack, expression.NewValuesFunc(er.ctx, col.Index, col.RetType))
 		return inNode, true
+	case *ast.WindowFuncExpr:
+		return er.handleWindowFunction(v)
 	default:
 		er.asScalar = true
 	}
 	return inNode, false
 }
 
+func (er *expressionRewriter) handleWindowFunction(v *ast.WindowFuncExpr) (ast.Node, bool) {
+	windowPlan, err := er.b.buildWindowFunction(er.p, v, er.aggrMap)
```

Comment:
```sql
SELECT val, 
       row_number()   OVER w AS 'row_number', 
       cume_dist()    OVER w AS 'cume_dist', 
       percent_rank() OVER w AS 'percent_rank' 
FROM numbers
window w AS (ORDER BY val);
```

Does the above query result in three window operators although they share the same window definition?

Reply:
Yes, merging windows will be done in the future.